core/arch/arm/kernel/rpc_io_i2c.c

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 Foundries Ltd <jorge@foundries.io>
+ */
+#include <kernel/rpc_io_i2c.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <string.h>
+
+/*
+ * @brief: i2c transfer request (read/write) via an adaptor from/to a device
+ * for a given length. It is the responsibility of the caller of this function
+ * to validate the number of bytes processed by the call.
+ *
+ * @param req:	mode: master transfer mode is either read or write.
```

Comment:
can remove this `req` structure description: already done in the header file.

Reply:
absolutely. more than happy to remove it.